tuf/client/updater.py

```diff
@@ -2584,11 +2594,11 @@ def targets_of_role(self, rolename='targets'):
     # Raise 'securesystemslib.exceptions.FormatError' if there is a mismatch.
     securesystemslib.formats.RELPATH_SCHEMA.check_match(rolename)
 
+    self._refresh_targets_metadata(refresh_all_delegated_roles=True)
+
```

Comment:
Why is refresh_all_delegated_roles necessary? This could be quite a few roles.


Next point is a bit complex:
`targets_of_role` remains a strange function to me if called on anything other than the Targets role itself. If the indicated role is a delegated role, it just isn't clear what it means to obtain trustworthy target info for whatever files listed by a trustworthy version of that role. (Whose word are we taking on the latter?)
Perhaps the best we can do is to: obtain some not-validated or partially-validated version of the delegated role (because delegation pathways depend on file paths and we haven't specified one), get the list of targets specified by that not-validated role file, and then properly obtain trustworthy target info for those file paths (whether or not they are ultimately provided by the same role?).

This more complex point plays into the correct way to run targets_of_role on a delegated role.

Reply:
My thinking is that `all_targets()` and `targets_of_roles()` should be for  clients that do not know ahead of time which targets are available on the repo, or provided by a particular role.

Community repos will certainly use get_one_valid_targetinfo(), but we might want to provide all_targets() and targets_of_roles() for simpler use cases / adopters.

Reply:
After an offline discussion, @awwad and I came to the conclusion that it's better to not support `all_targets()` and `targets_of_role()`.  We'll open a separate issue to remove these two functions.